### PR TITLE
feat(aws-cloudfront): Support Security Policy

### LINKED ISSUE #795: [CloudFront] Cannot set Security Policy
The current CloudFront implementation has no way as far as I can tell to set the Security Policy of the distribution (called MinimumProtocolVersion inside CloudFormation). This looks like it would fall under the AliasConfiguration object.

Docs can be seen here:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-viewercertificate.html#cfn-cloudfront-distribution-viewercertificate-minimumprotocolversion

### PR DESCRIPTION
**Pull request by ZeldoKavira**
Adds support for changing the default Security Policy (minimumProtocolVersion) with logic to ensure the proper one is being set for your SSLMethod. Fixes #795

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by ZeldoKavira** [comment]
@eladb || @rix0rrr Cannot set a reviewer

File: packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface AliasConfiguration {
   readonly names: string[],
   readonly acmCertRef: string,
   readonly sslMethod?: SSLMethod,
+  readonly securityPolicy?: SecurityPolicyProtocol
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: Can you add documentation here? I know it's not there for the other arguments, but improving the world has to start somewhere. Probably the documentation you put on the type can go here, and the type documentation can be more generic. And please pay attention to documenting the default using `@default`; what does it do if we leave it out?

Comment (ZeldoKavira) [reply packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: @rix0rrr Updated, put it in the header documentation with the others. Thanks for the feedback!

Comment (rix0rrr) [reply packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: Oh, that's actually not where they're supposed to go. But I'll fix it myself, feels unfair to make you do this chore.

**Comment by ZeldoKavira** [comment]
@rix0rrr Looks like this one is good to go now. The CONTRIBUTING docs say we can merge it but I do not have access.

**Pull request merged**